test/testfunctions.cpp

```diff
@@ -1975,6 +1975,12 @@ class TestFunctions : public TestFixture {
               "}\n");
         ASSERT_EQUALS("", errout.str());
 
+        check("void f(std::vector<std::unordered_map<int, std::unordered_set<int>>>& v, int i, int j) {\n"
+              "    auto& s = v[i][j];\n"
+              "    s.insert(0);\n"
```

Comment:
There should be test that the valueType of s is unordered_set.

Reply:
Turns out the ValueType was correct all along, but we don't use that when looking up member functions.
I have added a test where there is no ambiguity between containers.